homeassistant/components/alexa/flash_briefings.py

```diff
@@ -52,11 +55,24 @@ def get(self, request, briefing_id):
         """Handle Alexa Flash Briefing request."""
         _LOGGER.debug("Received Alexa flash briefing request for: %s", briefing_id)
 
-        if self.flash_briefings.get(briefing_id) is None:
+        if (
+            briefing_id == CONF_PASSWORD
+            or self.flash_briefings.get(briefing_id) is None
+        ):
             err = "No configured Alexa flash briefing was found for: %s"
             _LOGGER.error(err, briefing_id)
             return b"", HTTP_NOT_FOUND
 
+        if request.query.get(API_PASSWORD) is None:
+            err = "No password provided for Alexa flash briefing: %s"
+            _LOGGER.error(err, briefing_id)
+            return b"", HTTP_UNAUTHORIZED
+
+        if request.query[API_PASSWORD] != self.flash_briefings.get(CONF_PASSWORD):
```

Comment:
Use [`hmac.compare_digest`](https://docs.python.org/3/library/hmac.html#hmac.compare_digest) to avoid a timing attack.

Reply:
:+1: I will also move this check to the top to avoid timing attacks in the code you commented above.